src/core_modules/capture-core/components/WidgetsRelationship/common/RelationshipsWidget/useRelationshipTypes.js

```diff
@@ -23,7 +23,8 @@ export const useRelationshipTypes = (cachedRelationshipTypes?: RelationshipTypes
         relationshipTypesQuery,
         {
             enabled: !cachedRelationshipTypes?.length,
-            select: ({ relationshipTypes }: any) => relationshipTypes,
+            select: ({ relationshipTypes }: any) => relationshipTypes
+                .filter(({ access }) => access.data.write),
```

Comment:
See general comment first.

A better way to do this is to add filter to the api request, like so: http://localhost:9090/api/relationshipTypes?filter=access.data.read:eq:true